### ISSUE TITLE
Make VisualStudioDocumentTracker work in files that flow between having and not having ITextViews

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/Razor#2314]
In a model where a Razor file is opened without an `ITextView` and all of its Razor bits are created and then at a later time an `ITextView` is attached; we need to guard against the document tracker being subscribed multiple times. We almost do this today with our editor factory service that caches the document tracker on the text buffer but its only criteria on whether to subscribe or not is if there's a single text view.

Basically the flow that's broken is:
1. Index.cshtml is opened in the background by some system (live share) without a view.
2. Index.cshtml gets opened by the user, this attaches an `ITextView` to the file.

This results in the document tracker attaching to the project snapshot manager and all of its other info sources multiple times.
